sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -94,6 +94,7 @@ private[spark] case class SparkUserDefinedFunction(
     f: AnyRef,
     dataType: DataType,
     inputSchemas: Seq[Option[ScalaReflection.Schema]],
+    inputEncoders: Seq[ExpressionEncoder[_]] = Nil,
```

Comment:
Shall we have a simple `inputInfos: Seq[Option[(ScalaReflection.Schema, ExpressionEncoder)]]`?

Reply:
or is it possible to get the schema and nullability from the encoder?